coderd/httpmw/apikey.go

```diff
@@ -292,37 +306,51 @@ func ExtractAPIKey(rw http.ResponseWriter, r *http.Request, cfg ExtractAPIKeyCon
 				})
 			}
 
-			// If it is, let's refresh it from the provided config
+			if link.OAuthRefreshToken == "" {
+				return optionalWrite(http.StatusUnauthorized, codersdk.Response{
+					Message: SignedOutErrorMessage,
+					Detail:  fmt.Sprintf("%s session expired at %q.", friendlyName, link.OAuthExpiry.String()),
```

Comment:
Suggestion: Including an instruction in the detail for how to resolve would be helpful for the user.

Reply:
I think the best you can do it log in again.

Which the frontend should redirect you to the login page.